File: python/Binary Tree Inorder Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def inorderTraversal(self, root):
        self.ans = []
        self.doTraversal(root)
        return self.ans

    def doTraversal(self, node):
    	if node is None:
    		return
    	self.doTraversal(node.left)
    	self.ans.append(node.val)
    	self.doTraversal(node.right)

File: src/Binary Tree Inorder Traversal.cpp
//
//  main.cpp
//  test
//
//  Created by Septem on 3/29/14.
//  Copyright (c) 2014 Septem. All rights reserved.
//

#include <iostream>
#include <vector>
using namespace std;



//Definition for binary tree
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    vector<int> inorderTraversal(TreeNode *root) {
        ans.clear();
        doTraversal(root);
        return ans;
    }
    
    void doTraversal(TreeNode *node) {
        if (!node) {
            return;
        }
        doTraversal(node->left);
        ans.push_back(node->val);
        doTraversal(node->right);
    }
private:
    vector<int> ans;
};

int main(int argc, const char * argv[])
{

    // insert code here...
    std::cout << "Hello, World!\n";
    return 0;
}

